builder/context/buildcontext.go

```diff
@@ -73,7 +73,9 @@ func ForDirectory(dir string) (*BuildContext, error) {
 
 	directive, err := readDirectiveFile(fullDir)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to readDirectiveFile")
```

Comment:
note: 

I moved this out of the `readDirectiveFile` method to the build context because it felt out of place. 

IMO if the caller has expressed the intent to read the directive but it can't be found this is an error; it is the caller's (build context) responsibility to decide what they want to do in this event.